waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/DefaultMetaStoreClientFactory.java

```diff
@@ -132,6 +158,36 @@ private void reconnectIfDisconnected() {
       }
     }
 
+    private String genDelegationToken() throws Throwable {
+      base.open(cachedUgi);
+
+      UserGroupInformation currUser = UserGroupInformation.getCurrentUser();
+      if (delegationToken == null && currUser != UserGroupInformation.getLoginUser() && useSasl) {
+        String currShortName = currUser.getShortUserName();
+        Method getTokenMethod =
+                ThriftHiveMetastore.Iface.class.getMethod("get_delegation_token",
+                        String.class, String.class);
+        Object[] args = new Object[2];
+        args[0] = currShortName;
+        args[1] = currShortName;
+        String token = (String) doRealCall(getTokenMethod, args, 0);
+        base.close();
+        setTokenStr2Ugi(currUser, token);
+        this.delegationToken = token;
+        if (!base.isOpen()) {
+          base.open(cachedUgi);
+        }
+      }
+      return delegationToken;
+    }
+
+    private void setTokenStr2Ugi(UserGroupInformation currUser, String token) throws IOException {
+      String newTokenSignature = base.getHiveConfValue(HiveConf.ConfVars.METASTORE_TOKEN_SIGNATURE.varname,
```

Comment:
I'd suggest replacing the `getHiveConfValue` and `setHiveConfValue` methods could with a new method, something like `generateNewTokenSignature` which does all of this?